### PR TITLE
flag value bindings for kubectl attach/convert/delete/drain/edit/exec commands

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
xxxOptions did not get bound as default value for flags.
This PR cleans those flag bindings for `kubectl  attach/convert/delete/drain/edit/exec` commands.

/kind cleanup

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref #60366

**Special notes for your reviewer**:
/assign @deads2k 

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by dixudx** [comment]
@kubernetes/sig-cli-bugs PTAL. Thanks.

File: pkg/kubectl/cmd/convert.go
Comment (deads2k) [inline pkg/kubectl/cmd/convert.go]: Oh, this is neat.  Can you create a `NewDefaultConvertOptions() *ConvertOptions` function that does this?

File: pkg/kubectl/cmd/delete.go
Comment (deads2k) [inline pkg/kubectl/cmd/delete.go]: `NewDeleteOptions() *DeleteOptions` please

**Comment by deads2k** [comment]
@dixudx this is great!

@soltysh @juanvallejo keep an eye out for pulls like this. The cleanup will help us a bunch long term.

/approve

This is low risk.  I think it is safe to merge

/cc @kubernetes/sig-cli-maintainers

**Comment by deads2k** [comment]
> NewDeleteOptions() *DeleteOptions please

There are a couple others too.  Can you create the `New*` methods for them just to make construction and defaults a little more obvious.  lgtm otherwise.

**Comment by juanvallejo** [comment]
lgtm, pending creation of `New...Options` for option structs with custom default values

**Comment by dixudx** [comment]
@juanvallejo Updated. PTAL. Thanks

**Comment by deads2k** [comment]
Several of these have `out` and `errOut` as values in their structs.  I think the `NewFooOptions` methods should take that as input so that the relationship is obvious and the objects are valid.

/cc @juanvallejo

**Comment by juanvallejo** [comment]
@dixudx thanks for addressing the custom options approach.

I still see that "drain" and "edit" need to be updated as well.

```go
NewDrainOptions(f, out, errout)
```

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**
